src/peft/utils/safelora.py

```diff
@@ -0,0 +1,257 @@
+            The path of the LoRA weights and config.
+
+        select_layers_type (`Literal["threshold", "number"]`):
+            Specifies the method for selecting projection layers. The value must be a string and can only be either
+            "threshold" or "number". Use "threshold" to set a cosine similarity threshold or "number" to specify the
+            number of layers directly.
+
+        threshold (`float`):
+            The threshold of cosine similarity for selecting projected layers.
+
+        num_proj_layers (`int`):
+            The number of projected layers.
+
+        device (`str`):
+            Device that is used for SafeLoRA (cuda or cpu).
+
+        save_weights (`bool`):
+            Whether to override the original LoRA file with SafeLoRA weights.
+
+        local_files_only (`bool`):
+            Set this value to True to work only with local files (no downloads).
+
+        dtype (`torch.dtype`):
+            Data type for model weights, e.g., torch.float32 or torch.bfloat16.
+
+    """
+
+    base_model_path: str = field(
+        metadata={"help": "The path of the base model for obtaining the aligned matrix."},
+    )
+
+    aligned_model_path: str = field(
+        metadata={"help": "The path of the aligned model for obtaining the aligned matrix."},
+    )
+
+    peft_model_path: str = field(
+        metadata={"help": "The path of the LoRA wieghts and configs."},
+    )
+
+    select_layers_type: Literal["threshold", "number"] = field(
+        default="number",
+        metadata={"help": "How to select projection layers? options: [threshold, number]."},
+    )
+
+    threshold: float = field(
+        default=0.5,
+        metadata={"help": "The threshold of cosine similarity for selecting projected layers."},
+    )
+
+    num_proj_layers: int = field(
+        default=10,
+        metadata={"help": "The number of projected layers."},
+    )
+
+    device: str = field(
+        default="cuda",
+        metadata={"help": "Device that is used for SafeLoRA (cuda or cpu)."},
+    )
+
+    save_weights: bool = field(
+        default=True,
+        metadata={"help": "Whether to override the original LoRA file with SafeLoRA weights."},
+    )
+    local_files_only: bool = field(
+        default=False,
+        metadata={"help": "Set this value to True to work only with local files (no downloads)."},
+    )
+
+    dtype: torch.dtype = field(
+        default=torch.bfloat16,
+        metadata={
+            "help": "Data type for model weights, e.g., torch.float32 or torch.bfloat16. If your device is CPU, you should use torch.float32."
+        },
+    )
+
+    def __post_init__(self):
+        if self.base_model_path is None:
+            raise ValueError("base_model_path cannot be None.")
+        if self.aligned_model_path is None:
+            raise ValueError("aligned_model_path cannot be None.")
+        if self.peft_model_path is None:
+            raise ValueError("peft_model_path cannot be None.")
```

Comment:
Since there are no longer any default values for these fields, I don't believe we need to perform these checks anymore.